R/loadFromMothur.R

```diff
@@ -0,0 +1,206 @@
+      stop("'taxonomyTableFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    if(!is.null(sampleMetaFile) && !.is_non_empty_string(sampleMetaFile)){
+      stop("'sampleMetaFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    if(!.is_a_bool(featureNamesAsRefSeq)){
+      stop("'featureNamesAsRefSeq' must be TRUE or FALSE.", call. = FALSE)
+    }
+    if(!is.null(refSeqFile) && !.is_non_empty_string(refSeqFile)){
+      stop("'refSeqFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    if(!is.null(phyTreeFile) && !.is_non_empty_string(phyTreeFile)){
+      stop("'phyTreeFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    #
+    
+    # Reads the featureTablefile 
+    feature_tab <- .read_mothur_feature(featureTableFile, ...)
+    
+    # If rowData information exists, gets that. Otherwise, tax_tab is just data frame without information
+    if (!is.null(taxonomyTableFile)) {
+      taxa_tab <- .read_mothur_taxonomy(taxonomyTableFile, feature_tab, ...)
+    } else {
+      taxa_tab <- S4Vectors:::make_zero_col_DataFrame(nrow(feature_tab))
+    }
+    
+    # If colData information exists, gets that. Otherwise, sample_tab is just data frame without information
+    if (!is.null(sampleMetaFile)) {
+      sample_meta <- .read_mothur_sample_meta(sampleMetaFile, feature_tab)
+    } else {
+      sample_meta <- S4Vectors:::make_zero_col_DataFrame(ncol(feature_tab))
+    }
+    
+    if (!is.null(phyTreeFile)) {
+      tree <- .read_qza(phyTreeFile, ...)
+    } else {
+      tree <- NULL
+    }
+    
+    # if row.names(feature_tab) is a DNA sequence,  set it as refseq
+    if (!is.null(refSeqFile)){
+      refseq <- .read_qza(refSeqFile, ...)
+    } else if (featureNamesAsRefSeq) {
+      refseq <- .rownames_as_dna_seq(rownames(feature_tab))
+    } else {
+      refseq <- NULL
+    }
+    
+    return(TreeSummarizedExperiment(
+      assays = S4Vectors::SimpleList(counts = feature_tab),
+      rowData = taxa_tab,
+      colData = sample_meta,
+      rowTree = tree,
+      referenceSeq = refseq
+    ))
+    
+}
+
+.read_mothur_feature <- function(featureTableFile, ...){
+
+    # Reads the file
+    assay <- read.table(featureTableFile, check.names=FALSE, header=TRUE,
+                      sep="\t", stringsAsFactors=FALSE)
+  
+    # File contains additional columns "label", "numOtus", and "Group". They are removed
+    assay$label <- NULL
+    assay$numOtus <- NULL
+    # Information of "Group" column is saved, it includes taxa information
+    x <- assay$Group
+    assay$Group <- NULL
+    
+    # Initializes rownames
+    rownames(assay) <- NULL
+    # Saves taxa to rownames
+    rownames(assay) <- x
+    
+    # Transposes and converts assay from dataframe to matrix
+    assay <- t(as.matrix(assay))
+    
+    return(assay)
+    
```

Comment:
Please remove those empty lines. (All across the PR)

Reply:
Ok, should I avoid empty rows also in the future? Personally, I find it easier to read code when it is little bit grouped, but for others it might be confusing, unnecessary, or annoying

Reply:
That is a fine line. Grouping steps is helpful, but a break using `#` with a short description would work for that as well. An empty line after or before a return is not helpful. Have a look at the other code in `mia`. I think you will get it.